### ISSUE TITLE
[BUG] dart-code-metrics crash saying `Bad state: No element` when running command

### ISSUE THREAD

**Issue by fzyzcjy** [dart-code-checker/dart-code-metrics#804]
* **Dart code metrics version:** 4.14.0
* **Dart sdk version:** latest

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
n/a
```

</details>

**What did you do? Please include the source code example causing the issue.**

`flutter pub run dart_code_metrics:metrics analyze lib --fatal-style --fatal-performance --fatal-warnings --set-exit-on-violation-level noted`

```dart

```

**What did you expect to happen?**
no error

**What actually happened?**
```
[14:58:31|30. test_t.._proto]: ✔ no issues found!
[14:58:31|30. test_t.._proto]: Unhandled exception:
[14:58:31|30. test_t.._proto]: Bad state: No element
[14:58:31|30. test_t.._proto]: #0      IterableComparableExtension.max (package:collection/src/iterable_extensions.dart:744:5)
[14:58:31|30. test_t.._proto]: #1      maxMetricViolationLevel (package:dart_code_metrics/src/analyzers/lint_analyzer/utils/report_utils.dart:18:10)
[14:58:31|30. test_t.._proto]: #2      AnalyzeCommand.runCommand (package:dart_code_metrics/src/cli/commands/analyze_command.dart:83:9)
[14:58:31|30. test_t.._proto]: <asynchronous suspension>
[14:58:31|30. test_t.._proto]: #3      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
[14:58:31|30. test_t.._proto]: <asynchronous suspension>
[14:58:31|30. test_t.._proto]: #4      CliRunner.run (package:dart_code_metrics/src/cli/cli_runner.dart:51:7)
[14:58:31|30. test_t.._proto]: <asynchronous suspension>
[14:58:31|30. test_t.._proto]: #5      main (file:///Users/tom/.pub-cache/hosted/pub.flutter-io.cn/dart_code_metrics-4.14.0/bin/metrics.dart:4:3)
[14:58:31|30. test_t.._proto]: <asynchronous suspension>
[14:58:31|30. test_t.._proto]: pub finished with exit code 255
```

**Are you willing to submit a pull request to fix this bug?**

**Comment by incendial** [comment]
Available in 4.15.0 🚀

**Comment by fzyzcjy** [comment]
:tada
